### PR TITLE
Add support for kebab-case in doc-review

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/47536
fixes: https://github.com/demisto/demisto-sdk/issues/1978

## Description
Added support for sub words suggestion in kebab-case sentences when running the **doc-review** command.

### CODE REVIEW

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 1]
```diff
@@ -318,19 +318,23 @@ def remove_punctuation(word):
     def check_word(self, word):
         """Check if a word is legal"""
         # check camel cases
+        word = self.remove_punctuation(word)
+        sub_words = []
+        if '-' in word:
+            sub_words.extend(word.split('-'))
         if not self.no_camel_case and self.is_camel_case(word):
```
Comment (dorschw) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:325]: `elif`?

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py [hunk 2]
```diff
@@ -318,19 +318,23 @@ def remove_punctuation(word):
...
-                    self.unknown_words[word] = list(self.spellchecker.candidates(sub_word))[:5]
-
+            sub_words.extend(self.camel_case_split(word))
         else:
-            word = self.remove_punctuation(word)
-            if word.isalpha() and self.spellchecker.unknown([word]):
-                self.unknown_words[word] = list(self.spellchecker.candidates(word))[:5]
+            sub_words.append(word)
+
+        self.unknown_words[word] = []
```
Comment (dorschw) [inline demisto_sdk/commands/doc_reviewer/doc_reviewer.py:330]: perhaps a set would be better here?

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py (hunk 1 above)
Comment (DeanArbel) [reply demisto_sdk/commands/doc_reviewer/doc_reviewer.py:325]: I wasn't too sure about the validity of this case, but let's go with elif.

File: demisto_sdk/commands/doc_reviewer/doc_reviewer.py (hunk 2 above)
Comment (DeanArbel) [reply demisto_sdk/commands/doc_reviewer/doc_reviewer.py]: Had to make sure - this'll only change a print, so it's ok to change it to set.

**Review by dorschw — APPROVED** [review]

**Pull request merged**